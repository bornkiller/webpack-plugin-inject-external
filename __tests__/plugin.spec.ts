/* eslint-disable import/no-extraneous-dependencies, no-console */

// package
import fs from 'fs';
import path from 'path';
import webpack from 'webpack';
import MemoryFS from 'memory-fs';
import HtmlWebpackPlugin from 'html-webpack-plugin';

// Internal
import InjectExternalPlugin from '../src';

// Scope
const mfs = new MemoryFS();

describe('plugin test suits', () => {
  it('should complete standard workflow', (done) => {
    const target = path.resolve(
      __dirname,
      '__fixture__',
      'bootcdn.externals.yml'
    );
    const configuration = {
      entry: path.resolve(__dirname, '__fixture__', 'index.js'),
      output: {
        path: path.resolve(process.cwd(), 'dist'),
        filename: '[name].js',
        publicPath: '/',
      },
      externals: {
        moment: 'moment',
      },
      plugins: [
        new HtmlWebpackPlugin({
          template: path.resolve(__dirname, '__fixture__', 'index.html'),
          inject: 'body',
        }),
        new InjectExternalPlugin({
          env: 'production',
          definition: target,
        }),
      ],
    };
    const compiler = webpack(configuration);
    const outputPath = `${configuration.output.path}/index.html`;

    compiler.inputFileSystem = fs;
    compiler.outputFileSystem = mfs;

    compiler.run((err, stat) => {
      try {
        expect(err).toBeNull();
        expect(stat.hasErrors()).toBeFalsy();
        expect(mfs.readFileSync(outputPath, 'utf8')).toMatchSnapshot();
        done();
      } catch (error) {
        done(error);
      }
    });
  });
});
