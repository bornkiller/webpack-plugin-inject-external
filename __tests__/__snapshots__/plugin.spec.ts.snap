// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin test suits should complete standard workflow 1`] = `"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><title>Webpack Plugin</title></head><body><script integrity=\\"sha384-fYxN7HsDOBRo1wT/NSZ0LkoNlcXvpDpFy6WzB42LxuKAX7sBwgo7vuins+E1HCaw\\" src=\\"https://cdn.bootcss.com/moment.js/2.22.1/moment.min.js\\" crossorigin=\\"anonymous\\"></script><script integrity=\\"sha384-XjUTsP+pYBX4Kwg40gPvhWcRGKZh9gUpDForgg5bwsnyNU+VabKhRX6XkyM6fLk4\\" src=\\"https://cdn.bootcss.com/moment.js/2.22.1/locale/zh-cn.js\\" crossorigin=\\"anonymous\\"></script><script src=\\"/main.js\\"></script></body></html>"`;
